### PR TITLE
Add the PopupMonitor lib

### PR DESCRIPTION
**Pull request by scruffian**
This updates our usage of PopupMonitor to use a package exported from Calypso.

#### Changes proposed in this Pull Request:

* Use the Calypso package
* Update the API call to get the connect URL

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Update to existing feature

#### Testing instructions:

* Checkout this branch
* Run `yarn` to install the new package
* Add an Instagram Gallery to your post
* Click the connect button
* Check that the popup still opens

* Also navigate to your traffic settings: `/wp-admin/admin.php?page=jetpack#/traffic`
* Click on Verify With Google
* Check that a dialog pops up

#### Proposed changelog entry for your changes:

* Use PopupMonitor from Calypso

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
Let's take that opportunity to update the uses of `PopupMonitor` in the React interface as well, and get rid of our local `PopupMonitor` lib?
It seems it's only used in one place at the moment:
https://github.com/Automattic/jetpack/blob/36c23fa3bebc1d7c24d0aeb4e63d265c0dd97d12/_inc/client/lib/sharing/index.js#L4

File: _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ public function get_instagram_connect_url() {
 			return $response;
 		}
 		$body        = json_decode( wp_remote_retrieve_body( $response ) );
-		$connect_url = $body->services->instagram->connect_URL;
+		$connect_url = $body->services->{'instagram-basic-display'}->connect_URL;
```
Comment (jeherve) [inline _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram.php:102]: Is this an unrelated change?

Comment (Copons) [reply _inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram.php:102]: This is unrelated but also likely cause for your issue in the #15470 (see https://github.com/Automattic/jetpack/pull/15470#issuecomment-615308894).

I'm fine with having this here, as it would otherwise be impossible to test it.

I'll also make sure to rebase #15470 once this is in.

**Review by Copons — COMMENTED** [review]
This works for me!
I agree with @jeherve that we can update the other use of `PopupMonitor` and delete the old lib.

**Comment by Copons** [comment]
@scruffian the PopupMonitor package has been published!
https://www.npmjs.com/package/@automattic/popup-monitor

**Comment by scruffian** [comment]
I also removed the duplicated popup-monitor and referenced the package in the sharing section. The test instructions are updated.

**Review by Copons — DISMISSED** [review]
Tested both the Instagallery block and Verify with Google, the popup shows up (and closes) as expected.

Since the package is [published on npm](https://www.npmjs.com/package/@automattic/popup-monitor), it should be enough to just run `yarn` in the JP root to install it.

**Review by jeherve — APPROVED** [review]
This seems to work well. Should be good to merge once the tests pass (had to rebase to force Travis to come by the PR again).

**Comment by scruffian** [comment]
@copons we should be able to put the block on wpcom now...

**Comment by kraftbj** [comment]
r206616-wpcom
